src/components/MAPI/keypairs/ClusterDetailKeyPairs.tsx

```diff
@@ -142,23 +161,23 @@ const ClusterDetailKeyPairs: React.FC<IClusterDetailKeyPairsProps> = () => {
     setSelectedKeyPairSerial('');
   };
 
-  const provider = window.config.info.general.provider;
-
-  const organizations = useSelector(selectOrganizations());
-  const selectedOrg = orgId ? organizations[orgId] : undefined;
-  const namespace = selectedOrg?.namespace;
+  const { canGet: canGetCluster } = usePermissionsForClusters(
+    provider,
+    namespace ?? ''
+  );
 
-  const clusterKey = namespace
-    ? fetchClusterKey(provider, namespace, clusterId)
-    : null;
+  const clusterKey =
+    canGetCluster && namespace
+      ? fetchClusterKey(provider, namespace, clusterId)
+      : null;
 
   // The error is handled in the parent component.
   const { data: cluster } = useSWR<Cluster, GenericResponseError>(
     clusterKey,
     () => fetchCluster(clientFactory, auth, provider, namespace!, clusterId)
   );
 
-  const canCreateClientCertificates = cluster
+  const canCreateClientCertificatesForCluster = cluster
```

Comment:
Renamed this for accuracy, and to be less confusing with the permissions-related variables with similar names